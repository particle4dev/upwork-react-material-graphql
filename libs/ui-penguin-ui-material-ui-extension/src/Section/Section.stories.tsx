import * as React from 'react';
import Typography from '@material-ui/core/Typography';
import { withKnobs } from "@storybook/addon-knobs";
import Section from './Section';

export default {
  title: "Section",
  decorators: [withKnobs]
};

export const Basic = () => <div style={{backgroundColor: 'red'}}><Section><Typography>Basic</Typography></Section></div>;
