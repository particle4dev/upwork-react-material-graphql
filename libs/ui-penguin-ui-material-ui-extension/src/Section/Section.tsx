import * as React from 'react';
import classnames from 'classnames';
import { WithStyles, createStyles, withStyles, Theme } from '@material-ui/core';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const debug = require('debug')('@penguin-ui/material-ui-extension:Section');

export const styles = (theme: Theme) => createStyles({
  root: {
    position: 'relative',
    // minHeight: 500,
    padding: `0 ${theme.spacing(2)}px`,

    [theme.breakpoints.up('sm')]: {},

    [theme.breakpoints.up('md')]: {
      padding: `0 ${theme.spacing(4)}px`,
    },

    [theme.breakpoints.up('lg')]: {},
  }
});

export type SectionProps = WithStyles<typeof styles> & {
  children: React.ReactNode;
  id?: string;
  name?: string;
  className?: string;
  style?: React.CSSProperties;
}

const Section = React.forwardRef(function Section(props: SectionProps, ref: React.Ref<HTMLDivElement>) {
  debug('render');

  const {children, classes, id, className, ...other} = props;

  const cls = classnames(classes.root, className);

  return (
    <div data-id={props.id} data-name={props.name} className={cls} ref={ref} {...other}>
      {children}
    </div>
  );
});

if (process.env.NODE_ENV !== 'production') {
  Section.displayName = 'penguin_ui_material_ui_extension__Section';
}

Section.defaultProps = {};

export default withStyles(styles, { name: 'mui_extension__Section' })(Section);
