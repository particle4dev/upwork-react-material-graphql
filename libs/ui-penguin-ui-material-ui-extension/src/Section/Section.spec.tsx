import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Section from './Section';

describe('Section', () => {
  it('should render successfully', () => {
    const { baseElement, getByText } = render(<Section>test</Section>);
    expect(baseElement).toBeTruthy();
    expect(getByText('test')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
