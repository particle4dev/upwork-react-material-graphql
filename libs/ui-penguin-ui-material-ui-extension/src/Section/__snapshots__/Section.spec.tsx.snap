// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render successfully 1`] = `
<body>
  <div>
    <div
      class="mui_extension__Section-root-1"
    >
      test
    </div>
  </div>
</body>
`;
